src/GraphQL.Tests/Serialization/DeserializationTests.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Shouldly;
+using Xunit;
+
+namespace GraphQL.Tests.Serialization
+{
+    public class DeserializationTests : DeserializationTestBase
+    {
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void StringToInputs(IGraphQLTextSerializer serializer)
+        {
+            var actual = serializer.Deserialize<Inputs>(ExampleJson);
+            Verify(actual);
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void FromJson(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"query\":\"hello\",\"variables\":{ExampleJson}}}";
+            var actual = serializer.Deserialize<TestClass1>(test);
+            actual.Query.ShouldBe("hello");
+            Verify(actual.Variables);
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void FromJson_Null(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"query\":\"hello\",\"variables\":null}}";
+            var actual = serializer.Deserialize<TestClass1>(test);
+            actual.Query.ShouldBe("hello");
+            actual.Variables.ShouldBeNull();
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void FromJson_Missing(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"query\":\"hello\"}}";
+            var actual = serializer.Deserialize<TestClass1>(test);
+            actual.Query.ShouldBe("hello");
+            actual.Variables.ShouldBeNull();
+        }
+
+        [Theory]
+        [ClassData(typeof(GraphQLSerializersTestData))]
+        public void FromJson_IsCaseInsensitive_Element(IGraphQLTextSerializer serializer)
+        {
+            var test = $"{{\"Query\":\"hello\",\"Variables\":{ExampleJson}}}";
+            var actual = serializer.Deserialize<TestClass2>(test);
```

Comment:
So "Query" works for custom classes like `TestClass2` but does not for `GraphQLRequest`. Right?

Reply:
That is correct.  `GraphQLRequest` and `OperationMessage` are always case-sensitive and always use camelCase.  When serializing, they always serialize to camelCase, as do `ApolloTrace` and `ExecutionResult`.

I'm open to changing `GraphQLRequest` to be case-insensitive.  I figured we would start with it as-is and see if we have any complaints.  Perhaps we should note in the migration document.

Reply:
> Perhaps we should note in the migration document.

Oh, I already did.

Reply:
As is OK.